forastero/scoreboard.py

```diff
@@ -203,8 +213,28 @@ async def _dequeue(self) -> tuple[BaseTransaction, BaseTransaction]:
         await self._q_mon.wait_for_not_empty()
         # When a monitor transaction arrives, lock out the drain procedure
         await self._lock.acquire()
-        # Wait for the reference model to provide a transaction
-        next_ref = await self._q_ref.pop()
+        # If matching window is 1, wait for the first reference transaction
+        if self.match_window == 1:
+            next_ref = await self._q_ref.pop()
+        # If matching window > 1, peek at the captured transaction and search
+        # ahead the next N transactions
+        else:
+            peek_mon = self._q_mon[0]
+            next_ref = None
+            while next_ref is None:
+                # Search within the match window
+                for idx, peek_ref in enumerate(self._q_ref[: self.match_window]):
+                    if peek_ref == peek_mon:
+                        next_ref = await self._q_ref.pop(idx)
+                        break
```

Comment:
Is the below scenario possible:

There is a series of reference transactions that match an entry within the monitor queue window but never entry 0. Entry 0 remains in the queue for longer than the window length. Then a reference transaction arrives that matches monitor queue entry 0 even though it is outside the window.